core/services/head_tracker.go

```diff
@@ -250,7 +251,9 @@ func (ht *HeadTracker) Connected() bool {
 
 func (ht *HeadTracker) connect(bn *models.Head) {
 	for _, trackable := range ht.callbacks {
-		ht.logger.WarnIf(trackable.Connect(bn))
+		if err := trackable.Connect(bn); err != nil {
```

Comment:
Why not leave it as ht.logger.WarnIf? Have I missed something?

Reply:
The initial implementation was using a `zap.Logger`. After a code review it was changed back to using a `logger.Logger` so all the changes related to the `XXXIf` should be reverted